pkg/apis/core/validation/validation.go

```diff
@@ -2157,6 +2195,28 @@ func ValidatePersistentVolumeClaimSpec(spec *core.PersistentVolumeClaimSpec, fld
 	return allErrs
 }
 
+func isDataSourceEqualDataSourceRef(dataSource *core.TypedLocalObjectReference, dataSourceRef *core.TypedObjectReference) bool {
+	if dataSource == nil && dataSourceRef == nil {
+		return true
+	}
+	if dataSource == nil || dataSourceRef == nil {
```

Comment:
I changed it to use reflect.DeepEqual because of using the reflect package.

https://github.com/kubernetes/apimachinery/blob/master/pkg/api/equality/semantic.go#L29
https://github.com/kubernetes/apimachinery/blob/9e85d3af4ae2aff3b86c43092d1b0e5a6c942fef/pkg/conversion/deep_equal.go#L30
https://github.com/kubernetes/apimachinery/blob/9e85d3af4ae2aff3b86c43092d1b0e5a6c942fef/third_party/forked/golang/reflect/deep_equal.go#L202-L203